EasyPost.Tests/ServicesTests/BillingServiceTest.cs

```diff
@@ -0,0 +1,211 @@
+        }
+
+        [Fact]
+        [CrudOperations.Read]
+        [Testing.Function]
+        public async Task TestRetrievePaymentMethodsSummary()
+        {
+            UseMockClient();
+
+            PaymentMethodsSummary paymentMethodsSummary = await Client.Billing.RetrievePaymentMethodsSummary();
+
+            Assert.NotNull(paymentMethodsSummary.PrimaryPaymentMethod);
+            Assert.NotNull(paymentMethodsSummary.SecondaryPaymentMethod);
+        }
+
+        [Fact]
+        [Testing.Exception]
+        public async Task TestRetrievePaymentMethodsSummaryNoId()
+        {
+            UseMockClient(new List<TestUtils.MockRequest>
+            {
+                new(
+                    new TestUtils.MockRequestMatchRules(Method.Get, @"^v2\/payment_methods$"),
+                    new TestUtils.MockRequestResponseInfo(HttpStatusCode.OK, data: new PaymentMethodsSummary
+                    {
+                        Id = null // No ID, will throw an error when we try to interact with this summary
+                    })
+                )
+            });
+
+            await Assert.ThrowsAsync<InvalidObjectError>(async () => await Client.Billing.RetrievePaymentMethodsSummary());
+        }
+
+        [Fact]
+        [CrudOperations.Delete]
+        [Testing.Function]
+        public async Task TestDeletePaymentMethod()
+        {
+            UseMockClient();
+
+            Exception? possibleException = await Record.ExceptionAsync(async () => await Client.Billing.DeletePaymentMethod(PaymentMethod.Priority.Primary));
+
+            Assert.Null(possibleException);
+        }
+
+        [Fact]
+        [CrudOperations.Delete]
+        [Testing.Logic]
+        public async Task TestGetPaymentMethodByPrioritySwitchCase()
+        {
+            UseMockClient();
+
+            // Deleting a payment method gets the payment method internally, which should test the switch case.
+            await Client.Billing.DeletePaymentMethod(PaymentMethod.Priority.Primary);
+            // The payment method is not exposed by this method, so we can't assert against it. If this test doesn't throw an exception, it worked (see test below)
+
+            // This time the internal switch case should use the secondary payment method. Again, if there's no exception thrown, it worked.
+            await Client.Billing.DeletePaymentMethod(PaymentMethod.Priority.Secondary);
+
+            // Now if we pass in a bad priority level, it should throw an exception as the default for the switch case
+            // Because this method only accepts enums (well, the custom enum class, but one that can't be extended/modified), we can't actually pass in a bad value.
+            // Therefore, something would have to be wildly unpredictable and horrible for this to actually error out for an end user.
+            // await Assert.ThrowsAsync<InvalidObjectError>(async () => await Client.Billing.DeletePaymentMethod(PaymentMethod.Priority.None));
```

Comment:
Shouldn't this be uncommented?

Reply:
No, it should be commented out. But I'll just remove it instead.